conda_smithy/configure_feedstock.py

```diff
@@ -376,6 +376,11 @@ def finalize_config(config, platform, forge_config):
             config["docker_image"] = [config["docker_image"][0]]
         else:
             config["docker_image"] = [forge_config["docker"]["fallback_image"]]
+
+    if "private_upload" in config:
+        config["private_upload"] = config["private_upload"]
+    elif "private_upload" in forge_config:
+        config["private_upload"] = forge_config["private_upload"]
```

Comment:
I am confused by this. Why are we stashing private upload in the ci support scripts?

We should leave this in the conda-forge,yml and use the value from there.

Reply:
I was trying to do the same as for `docker_image` but I got your point.

Reply:
Hmmmm. I did not add the docker_image stuff. @conda-forge/core why is docker image in the ci support data? Is that because it varies per platform?

Reply:
Yes according to the `cuda` flag I think.

Reply:
It should be solved now.

Reply:
Ok. My point here si that we don't need to set it in the config at all. The conda-forge-ci-support package could/should look directly at the conda-forge.yml. The ci support files are for stuff that ends up being used by conda build for building packages.

Reply:
Oh ok, I get your point. Then I need some guidance for how to do that in https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/98/files

Reply:
I tried to find the logic in `conda-forge-ci-support` that gets and uses the `docker_image` flag but cannot find it. The only place where the `docker_image` is used is the `conda-smithy` file.

Reply:
Right. Here is a variation that is slightly better than what I just said.

1. put the flag in the conda-forge.yml
2. use that flag value to render recipes with a `--private` flag passed to the ci support scripts or w/e that flag should be
3. handle the `--private` flag in the ci support scripts.

Reply:
Sorry but it looks very similar to what I am doing right now... `config["private_upload"]` is used to populate `ci_support` script. Then the PR in ` conda-forge-ci-support` handles the flag if present.

Could you explain in more detail what you want me to do?

Reply:
I might be wrong, but I think you need to populate the CI templates in the templates directory.

Reply:
But there is no template for the `ci_support` scripts.

Reply:
Unless I am missing something...

Reply:
I the actual azure/travis/etc templates etc here: https://github.com/conda-forge/conda-smithy/tree/master/conda_smithy/templates

Reply:
Ok so I need to message to https://github.com/conda-forge/conda-smithy/blob/master/conda_smithy/templates/build_steps.sh.tmpl instead of populating the `.ci_support/` scripts. Will do.

Reply:
Right except that only does linux. You need to handle windows and osx too.

Reply:
The current solution is much less verbose than what you propose. Are you sure about that? I don't understand the rationale behind it. I also don't the codebase very well so sorry if this is a naive question.

Reply:
My general feeling is that putting keys in ci support that are not generated by packages or our pinning files could lead to conflicts or errors. Those files are for conda build to use not for smithy.

Reply:
I’m happy to defer to other members of @conda-forge/core on this though.

Reply:
Ok, I understand. Thanks for the explanation.